File: C/Array_to_Tree.c
TreeNode* arrayToBinaryTree(int arr[], int* index, int size) {
    if (*index >= size || arr[*index] == -1) {
        (*index)++;
        return NULL;
    }

    TreeNode* node = (TreeNode*)malloc(sizeof(TreeNode));
    node->data = arr[(*index)++];
    node->left = arrayToBinaryTree(arr, index, size);
    node->right = arrayToBinaryTree(arr, index, size);
    return node;
}

void inOrderTraversal(TreeNode* root) {
    if (root) {
        inOrderTraversal(root->left);
        printf("%d ", root->data);
        inOrderTraversal(root->right);
    }
}

File: Python/Array_to_Tree.py
class TreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def array_to_binary_tree(arr, index):
    if index[0] >= len(arr) or arr[index[0]] == -1:
        index[0] += 1
        return None

    node = TreeNode(arr[index[0]])
    index[0] += 1
    node.left = array_to_binary_tree(arr, index)
    node.right = array_to_binary_tree(arr, index)
    return node

def in_order_traversal(root):
    if root:
        in_order_traversal(root.left)
        print(root.data, end=" ")
        in_order_traversal(root.right)
